packages/@best/api-db/src/sql/postgres/migrations/002-projects.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+*/
+
+export const up = `
+CREATE TABLE projects (
+    id SERIAL PRIMARY KEY,
+    name character varying(100) NOT NULL,
+    organization_id integer NOT NULL REFERENCES organizations(id) ON DELETE CASCADE,
+    name character varying(100) NOT NULL,
+    created_at timestamp without time zone NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    last_release_date timestamp without time zone
+);
+
+CREATE UNIQUE INDEX projects_unique_name ON projects(name text_ops);
+`
+
+export const down = `DROP TABLE projects;`
```

Comment:
This file is where I would expect _all_ the database migrations to live...

Reply:
i did not get you ,can you help me to understand more

Reply:
Sure, what I'm saying is this:

1. Let's say there's an existing database that has already run `001-organizations.ts`, but it ran the _old_ version.
2. Now when you upgrade the code and run migrations, the database will believe it has already run `001-organizations.ts`, so it will skip to `002-projects.ts`
3. However, since it didn't run the _updated_ `001-organizations.ts`, it fails because the `projects` table was not renamed to `organizations`.

I believe the standard approach with DB migrations is to never modify old migration files – only add new migration files.

Reply:
@nolanlawson  when it runs the best code it will check  for the organization table exist  ,if not  it will create new organization

Reply:
@sutturu I'm confused. It looks to me like, in the scenario I mentioned, it will run `002` but not `001`, the `projects` table will already exist, and so the `CREATE TABLE projects` command will fail.

Is this not the case? Also, what do you mean by "the best code"?